### ISSUE TITLE
Dockerng 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by valentin2105** [saltstack/salt#35758]
### Description of Issue/Question
Hi,

I want to ensure a docker image is present on the server with this State : 
```
mariadb:latest:
  dockerng.image_present
```

Here is the error on the Highstate :
```
An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1733, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1651, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/dockerng.py", line 551, in image_present
                  all_tags = __salt__['dockerng.list_tags']()
                File "/usr/lib/python2.7/dist-packages/salt/modules/dockerng.py", line 2253, in list_tags
                  for repo_tag in item.get('RepoTags', []):
              TypeError: 'NoneType' object is not iterable
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

Try to specify an "image_present" with Salt.

### Versions Report
```
Salt Version:
           Salt: 2016.3.2

Dependency Versions:
           cffi: 1.5.2
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: debian 8.5
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.5
```

**Comment by Ch3LL** [comment]
@valentin2105 I believe this to be a duplicate of #35563 but to make sure can you see if this fix works for you: #35447 

This behavior has been showing up if you have an image with teh <none> tag.

**Comment by valentin2105** [comment]
Yes it is the same problem. Sorry for the duplicate issue.
